docs/nuxt.config.js

```diff
@@ -12,9 +13,32 @@ renderer.code = (code, language) => {
   return `<pre class="hljs ${language} text-monospace p-2">${highlighted}</pre>`
 }
 
+// Custom heading implementation for markdown renderer
+// @link: https://github.com/nuxt/docs/blob/967fc39b4dc0712d2d5089014eddc7e7a2e65422/api.js#L27
+// @link: https://github.com/markedjs/marked/blob/1f5b9a19f532e2e1e3e63ae5efd81af75acf572f/lib/marked.js#L962
+renderer.heading = function(text, level, raw, slugger) {
+  let anchor = ''
+  if (this.options.headerIds) {
+    const pattern = /\s?{([^}]+)}$/
+    let link = pattern.exec(text)
+
+    if (link && link.length && link[1]) {
+      text = text.replace(pattern, '')
+      link = link[1]
+    } else {
+      link = this.options.headerPrefix + slugger.slug(raw)
+    }
+
+    const icon = octicons.link.toSVG()
+    anchor = `<a id="${link}" class="anchor" aria-hidden="true" href="#${link}">${icon}</a>`
+  }
+
+  return `<h${level}>${anchor}${text}</h${level}>\n`
```

Comment:
Shoudn't the ID be added to the `<h#>`, such as `<h{$level} id="${link}" ... </h${level}>` instead of on the `<a>` tag? (since the header element is a block level element with padding/margins while anchor tag is an inline style element lacking padding/etc).  This will help the scrollTo handlers properly scroll to the heading (offset by the padding/margins).

Reply:
Also, since the ID is no longer on the heading tag, the pages table of contents (and the search) no longer work, as the back end processor for that expects the ID's to be on the heading tags.

Reply:
Goode catch, will change that.